sema4ai/src/sema4ai_code/robocorp_language_server.py

```diff
@@ -1047,6 +1048,42 @@ def m_list_dev_tasks(self, action_package_uri: str) -> partial:
             partial(self._list_dev_tasks_in_thread, action_package_uri)
         )
 
+    def m_get_external_api_url(self) -> str | None:
+        home = get_default_sema4ai_home()
+        external_api_pid = home / "sema4ai-studio" / "external-api-server.pid"
+
+        if not external_api_pid.exists():
+            log.warning(
+                "Warning: Studio is not opened, cannot set the external API URL."
+            )
+            return None
+
+        with external_api_pid.open("r", encoding="utf-8") as file:
+            import json
+
+            file_content = json.load(file)
+
+        def _is_port_open(port) -> bool:
+            import socket
+
+            try:
+                conn = socket.create_connection(("localhost", port), timeout=1)
```

Comment:
I'd make the timeout even shorter (something like 0.01)... if it's there it should work instantly, if it's not, waiting more isn't helpful.